controllers/db/reports/publication.report.controller.ts

```diff
@@ -0,0 +1,754 @@
+});
+models.AnalysisSummaryArticle.hasOne(models.AnalysisSummaryAuthor, {
+  constraints: false,
+});
+
+export const generatePubsRtf = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  try {
+    let apiBody: GeneratePubsApiBody = JSON.parse(req.body);
+    let generatePubsRtfOutput: any = [];
+    console.log('coming here*****************',apiBody.personIdentifiers.length);
+    if (apiBody.personIdentifiers && apiBody.personIdentifiers.length > 0) {
+      generatePubsRtfOutput = await sequelize.query(
+        "CALL generatePubsRTF (:uids , :pmids)",
+        {
+          replacements: { uids: apiBody.personIdentifiers.join(','), pmids: apiBody.pmids.join(',') },
+          raw: true,
+        }
+      );
+    } else {
+
+      console.log('coming into this else one*************************')
+      generatePubsRtfOutput = await sequelize.query(
+        "CALL generatePubsNoPeopleRTF ( :pmids)",
+        {
+          replacements: { pmids: apiBody.pmids.join(',') },
+          raw: true,
+        }
+      );
+    }
+    
+    return Array.isArray(generatePubsRtfOutput) &&
+      generatePubsRtfOutput.length > 0
+      ? generatePubsRtfOutput[0].x
+      : "Error creating the file";
+  } catch (e) {
+    console.log(e);
+    res.status(500).send(e);
+  }
+};
+
+export const generatePubsPeopleOnlyRtf = async (
+    req: NextApiRequest,
+    res: NextApiResponse
+  ) => {
+    try {
+      let apiBody: GeneratePubsPeopleOnlyApiBody = JSON.parse(req.body);
+      const generatePubsPeopleOnlyRtfOutput: any = await sequelize.query(
+        "CALL generatePubsPeopleOnlyRTF (:uids)",
+        {
+          replacements: { uids: apiBody.personIdentifiers?apiBody.personIdentifiers.join(','):'' },
+          raw: true,
+        }
+      );
+      console.log(generatePubsPeopleOnlyRtfOutput)
+      return Array.isArray(generatePubsPeopleOnlyRtfOutput) &&
+      generatePubsPeopleOnlyRtfOutput.length > 0
+        ? generatePubsPeopleOnlyRtfOutput[0].x
+        : "Error creating the file";
+    } catch (e) {
+      console.log(e);
+      res.status(500).send(e);
+    }
+  };
+
+  export const generateAuthorshipReportCSV = async (
+    req: NextApiRequest,
+    res: NextApiResponse
+  ) => {
+    try {
+      let apiBody: PublicationSearchFilter = req.body;
+      const where = {};
+      var isAuthorFilters = false;
+      var isArticleFilters = false;
+      let isPersonTypeFilter = false;
+
+
+      if (apiBody.filters) {
+        where[Op.and] = [];
+        if (
+          apiBody.filters.journalTitleVerbose &&
+          apiBody.filters.journalTitleVerbose.length > 0
+        ) {
+          where[Op.and].push({
+            "$AnalysisSummaryArticle.journalTitleVerbose$": {
+              [Op.in]: apiBody.filters.journalTitleVerbose,
+            },
+          });
+        }
+        if (
+          apiBody.filters.personIdentifers &&
+          apiBody.filters.personIdentifers.length > 0
+        ) {
+          isAuthorFilters = true;
```

Comment:
so even though your setting this doesn't actually look like its being used anywhere